src/db/commit.ts

```diff
@@ -380,161 +359,168 @@ type Ref = {
 };
 
 async function newImpl(
-  builder: flatbuffers.Builder,
-  basisHash: Ref | undefined,
-  unionType: MetaTypedFB,
-  unionValue: number,
+  basisHash: Ref | null,
+  meta: LocalMeta | SnapshotMeta | IndexChangeMeta,
   valueHash: Ref,
-  originalHash: Ref | undefined,
+  originalHash: Ref | null,
   indexes: IndexRecord[],
 ): Promise<Commit> {
-  const meta = MetaFB.createMeta(
-    builder,
-    basisHash ? builder.createString(basisHash.h) : 0,
-    unionType,
-    unionValue,
-  );
-
-  const indexRecordsFB = [];
-  for (const index of indexes) {
-    const indexDefinition = IndexDefinitionFB.createIndexDefinition(
-      builder,
-      builder.createString(index.definition.name),
-      IndexDefinitionFB.createKeyPrefixVector(
-        builder,
-        utf8.encode(index.definition.keyPrefix),
-      ),
-      builder.createString(index.definition.jsonPointer),
-    );
-
-    const indexRecord = IndexRecordFB.createIndexRecord(
-      builder,
-      indexDefinition,
-      builder.createString(index.valueHash),
-    );
-    indexRecordsFB.push(indexRecord);
-  }
-
-  const commitFB = CommitFB.createCommit(
-    builder,
-    meta,
-    builder.createString(valueHash.h),
-    CommitFB.createIndexesVector(builder, indexRecordsFB),
-  );
-
-  builder.finish(commitFB);
-  const data = builder.asUint8Array();
-
-  const refs: (Ref | undefined)[] = [valueHash, basisHash, originalHash];
+  const refs: (Ref | null)[] = [valueHash, basisHash, originalHash];
   const strongRefs = (
     refs.filter(r => r && r.t === RefType.Strong) as Ref[]
   ).map(r => r.h);
   for (const index of indexes) {
     strongRefs.push(index.valueHash);
   }
 
+  const data: CommitData = {
+    meta,
+    valueHash: valueHash.h,
+    indexes,
+  };
   const chunk = Chunk.new(data, strongRefs);
   return new Commit(chunk);
 }
 
-function validate(data: Uint8Array) {
-  const buf = new flatbuffers.ByteBuffer(data);
-  const root = CommitFB.getRootAsCommit(buf);
-  if (root.valueHash() === null) {
-    throw new Error('Missing value hash');
-  }
-  const meta = root.meta();
-  if (!meta) {
-    throw new Error('Missing meta');
-  }
-
-  // basis_hash is optional -- the first commit lacks a basis
+export type CommitData = {
```

Comment:
@aboodman I was considering using arrays instead.

Pros (with Arrays instead of Objects):
- Order is predictable (for the tests it was important to add the properties in the same order. Not really a big deal for non tests)
- Less data to store
- No field names. With objects we need field names which cannot be renamed.
Cons:
- Less readable code. Needs getters in code to make code.
- Removing a field would required keeping a "hole" in the array.

Reply:
feels intuitively right to me to just use objects. can't really say why... easier to develop i guess.